### PR TITLE
Add loose strategy for missing plates in MCMC

### LINKED ISSUE #1276: [FR] raise missing plate warning in MCMC inference
basically extend #1245 to be enforced more globally

### PR DESCRIPTION
**Pull request by fehiepsi**
Fixes #1276

### CODE REVIEW

**Review by martinjankowiak — APPROVED** [review]
lgtm. this has no effect on the current svi behavior, right?

**Comment by fehiepsi** [comment]
Yes, but we can merge two implementations. The svi version is better because it also handles singleton dimensions. Let's me merge them.

**Review by martinjankowiak — DISMISSED** [review]
lgtm. @fehiepsi is this still wip?

**Comment by fehiepsi** [comment]
Thanks for reviewing, @martinjankowiak! I just fixed some tests due to new jaxlib/scikit-learn release:
```
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/sklearn/utils/deprecation.py:87: 
FutureWarning: Function get_feature_names is deprecated; get_feature_names is deprecated in 1.0 and will be removed in 1.2. Please use get_feature_names_out instead.
```

File: numpyro/infer/elbo.py [hunk 1]
```diff
@@ -569,6 +574,8 @@ def single_particle_elbo(rng_key):
             model_trace, guide_trace = get_importance_trace(
                 seeded_model, seeded_guide, args, kwargs, param_map
             )
+            check_model_guide_match(model_trace, guide_trace)
+            _validate_model(model_trace, plate_warning="strict")
```
Comment (martinjankowiak) [inline numpyro/infer/elbo.py:578]: why strict here?

**Review by martinjankowiak — APPROVED** [review]

File: numpyro/infer/elbo.py (hunk 1 above)
Comment (fehiepsi) [reply numpyro/infer/elbo.py:578]: I'm seeing TraceGraphELBO leverages plate stacks etc to do inference. I'm not sure if strict is required. If you think using loose is enough, I can make the change.

Comment (fehiepsi) [reply numpyro/infer/elbo.py:578]: Let me merge this first, then would make the change if needed in #1310

**Pull request merged**